scripts/results_generation/generate_reproductions.py

```diff
@@ -65,76 +57,17 @@ def main():
-                        [0, 0, 1, 0, 0, 0],
-                        [0, 0, 0, 1, 0, 0],
-                        [0, 0, 0, 0, 1, 0],
-                        [magnitude, magnitude, magnitude, magnitude, magnitude, 1],
-                    ]
-                )
-                target = np.matmul(target, translation_matrix)
-                initial_state = np.matmul(initial_state, translation_matrix)
-            elif j == 1:
-                # apply rotation on both initial and target references
-                angle_rad = np.random.uniform(-math.pi, math.pi)
-                rotation_matrix = np.array(
-                    [
-                        [np.cos(angle_rad), np.sin(angle_rad), 0, 0, 0, 0],
-                        [-np.sin(angle_rad), np.cos(angle_rad), 0, 0, 0, 0],
-                        [0, 0, 1, 0, 0, 0],
-                        [0, 0, 0, np.cos(angle_rad), np.sin(angle_rad), 0],
-                        [0, 0, 0, -np.sin(angle_rad), np.cos(angle_rad), 0],
-                        [0, 0, 0, 0, 0, 1],
-                    ]
-                )
-                target = np.matmul(target, rotation_matrix)
-                initial_state = np.matmul(initial_state, rotation_matrix)
-            elif j == 2:
-                # apply scaling on both initial and target references
-                scaling_factor = np.random.uniform(0, 2)
-                scaling_matrix = np.array(
-                    [
-                        [scaling_factor, 0, 0, 0, 0, 0],
-                        [0, scaling_factor, 0, 0, 0, 0],
-                        [0, 0, scaling_factor, 0, 0, 0],
-                        [0, 0, 0, scaling_factor, 0, 0],
-                        [0, 0, 0, 0, scaling_factor, 0],
-                        [0, 0, 0, 0, 0, scaling_factor],
-                    ]
-                )
-                target = np.matmul(target, scaling_matrix)
-                initial_state = np.matmul(initial_state, scaling_matrix)
-            elif j == 3:
-                # apply noise on initial reference (simulate fixed target)
-                noise_i = 0.2 * np.random.rand(6)
-                target = regression.prediction[-1, 1:]
-                initial_state = apply_noise(target, noise_i)
-            elif j == 4:
-                # apply noise on target reference (simulate fixed initial position)
-                noise_t = 0.2 * np.random.rand(6)
-                target = apply_noise(target, noise_t)
-                initial_state = regression.prediction[0, 1:]
-            elif j == 5:
-                # apply noise on both initial and target references
-                noise_b = 0.2 * np.random.rand(6)
-                target = apply_noise(target, noise_b)
-                initial_state = apply_noise(initial_state, noise_b)
+            # apply noise on both initial and target references
+            noise = np.random.normal(scale=std_dev)
+            noise_vector = noise * np.ones(np.shape(target))
```

Comment:
I think you only need `target = target + noise` without having to make it a noise vector, no :)?

Reply:
True!